ci/requirements-3.7-dev.yml

```diff
@@ -25,7 +25,7 @@ dependencies:
   - plumbum
   - pre-commit
   - psycopg2
-  - pyarrow>=0.13
+  - pyarrow>=0.13, <=0.14.1
```

Comment:
really? why do we actually need this? i would rather move up the pyarrow >= 0.14 (or even higher)

Reply:
I still don't know specifically why, but some tests in `test_temporal.py` fail if pyarrow is 0.15.0 higher and succeed if pyarrow is 0.14.1 or lower.

pyarrow used to happen to solve to 0.13.0, but unrelated changes in requirements.yml ([in this commit](https://github.com/ibis-project/ibis/commit/3d9756cee1dc3e0c622891dd6314c2f0abecc382) inadvertently made pyarrow solve to 0.17.1 (if `pymapd` is commented out from the requirements yml), or to 0.15.0 (if `pymapd` is enabled in the requirements yml)

I could still try separate requirements .yml files for each backend set (would have to split omnisci and PySpark apart because of the requirements conflicts). But it sounds like maybe it's worth it for me to see if the failing tests themselves can be changed so we don't have to limit the version

Reply:
so i am ok on *some* builds to use older versions, but what I would suggest is that we create a new .yml for *just* spark (you will have to separate it from the omnisci backend) and just go from there to get this working.

Reply:
The newest Spark release (2.4) depends on Arrow 0.10 on the java side. It's possible that it's too old and doesn't work with newer version of PyArrow.

Reply:
ok fair enough. let's get this working, then create an issue about using the latest versions (or simply change the supported versions).